python/tests/test_mesh.py

```diff
@@ -25,9 +32,12 @@ def test_immutable(mesh):
         mesh.step = 1.0
 
 def test_volume_cartesian(cartesian_mesh):
+    assert cartesian_mesh.lower_boundary_condition ==  cartesian_mesh.boundary_type.periodic
+    assert cartesian_mesh.upper_boundary_condition ==  cartesian_mesh.boundary_type.periodic
     assert cartesian_mesh.volume() == pytest.approx(10.)
     assert cartesian_mesh.volume(0) == pytest.approx(0.5)
 
 def test_volume_spherical(spherical_mesh):
+    assert spherical_mesh.lower_boundary_condition ==  spherical_mesh.boundary_type.reflect
```

Comment:
```suggestion
    assert spherical_mesh.lower_boundary_condition ==  "reflect"
```
You also need to test the upper bc.